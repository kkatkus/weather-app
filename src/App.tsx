import React from 'react';
import { useSelector } from 'react-redux';
import { ThemeProvider } from 'emotion-theming';

import AppRoutes from './AppRoutes';
import RootState from './RootState';
import * as themes from './shared/styles';
import Layout from './features/layout';
import Notification from './shared/components/Notification';

const App = () => {
  const [theme] = useSelector((state: RootState) => [state.settings.theme]);
  return (
    <ThemeProvider theme={themes[theme]}>
      <Layout>
        <AppRoutes />
        <Notification />
      </Layout>
    </ThemeProvider>
  );
};

export default App;
