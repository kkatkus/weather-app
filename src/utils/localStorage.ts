export const isLocalStorageAvailable = (): boolean => {
  const test = 'test';
  try {
    localStorage.setItem(test, test);
    localStorage.removeItem(test);
    return true;
  } catch (e) {
    return false;
  }
};

export const getItem = (key: string) => {
  if (!isLocalStorageAvailable() || !key) {
    return undefined;
  }
  const data = localStorage.getItem(key);
  try {
    return data ? JSON.parse(data) : null;
  } catch (e) {
    console.group('localStorage - getItem - Failed to parse string to json');
    console.log(data);
    console.log(e.message);
    console.groupEnd();
    return null;
  }
};

export const setItem = (key: string, data: number | string | boolean | object | null | undefined): void => {
  if (!isLocalStorageAvailable() || !key) {
    return;
  }
  localStorage.setItem(key, JSON.stringify(data));
};

export const removeItem = (key: string): void => {
  if (!isLocalStorageAvailable() || !key) {
    return;
  }
  localStorage.removeItem(key);
};
