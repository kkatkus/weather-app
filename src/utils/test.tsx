import React, { ReactNode } from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mergeDeepRight } from 'ramda';

import { ThemeProvider } from 'emotion-theming';
import * as themes from '../shared/styles';
import RootState from '../RootState';

const defaultState = {
  weather: {
    loading: false,
    error: '',
    place: {},
    observation: {},
    history: [],
  },
  settings: {
    loading: false,
    theme: 'light',
  },
};

export const getMockProvider = (partialState: any = {}) => {
  const mockStore: any = configureStore();
  const store: any = mockStore(mergeDeepRight(defaultState, partialState));

  const MockProvider = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <ThemeProvider theme={themes[store.getState().settings.theme]}>{children}</ThemeProvider>
    </Provider>
  );

  return {
    MockProvider,
    store,
  };
};
