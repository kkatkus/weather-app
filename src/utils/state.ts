import { STATE_KEY_NAME } from '../constants';

import { getItem, setItem } from './localStorage';

export const getState = () => {
  const state = getItem(STATE_KEY_NAME);
  if (state === null) {
    return undefined;
  }
  return state;
};

export const saveState = (state: object): void => {
  setItem(STATE_KEY_NAME, state);
};
