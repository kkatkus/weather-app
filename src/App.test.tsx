import React from 'react';
import { shallow } from 'enzyme';

import App from './App';
import { MemoryRouter } from 'react-router';
import { getMockProvider } from './utils/test';

describe('<App />', () => {
  it('matches snapshot', () => {
    const { MockProvider } = getMockProvider();
    const wrapper = shallow(
      <MockProvider>
        <MemoryRouter initialEntries={[{ pathname: '/weather', key: 'ratesKey' }]}>
          <App />
        </MemoryRouter>
      </MockProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
