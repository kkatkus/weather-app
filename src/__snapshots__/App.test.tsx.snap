// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "background": "#FFFFFF",
          "boxShadow": "0 0 0.1rem rgba(0, 0, 0, 0.08), 0 0.1rem 0.1rem rgba(0, 0, 0, 0.14)",
          "error": "rgb(235, 0, 141)",
          "header": "#FFFFFF",
          "input": "rgb(139, 149, 158)",
          "inputBg": "rgba(0,0,0,0.025)",
          "logo": "transparent",
          "onBackground": "#313131",
          "onError": "#FFFFFF",
          "onInput": "rgb(25, 28, 31)",
          "onPrimary": "#FFFFFF",
          "onSecondary": "#FFFFFF",
          "primary": "rgb(235, 0, 141)",
          "secondary": "#0075eb",
        },
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "ratesKey",
            "pathname": "/weather",
          },
        ]
      }
    >
      <App />
    </MemoryRouter>
  </ThemeProvider>
</Provider>
`;
